### PR TITLE
[#9585] No user feedback for empty search results on Admin page

### LINKED ISSUE #9585: No user feedback for empty search results on Admin page
- **Environment**: 
`master` branch at commit `9542ad8`.

**Steps to reproduce**
1. Log in as Admin
2. Go to Admin Search page
3. Search for some random string (which will not return any results from the backend)

**Expected behaviour**
There is some message informing the user that no results were found

**Actual behaviour**

There is no feedback to the user:
2019-03-19_23h36_04

**Additional Notes**
The instructor search page displays a relevant message at the top of the page
2019-03-19_23h40_23

### PR DESCRIPTION
**Pull request by Davileal**
Fixes #9585 

**Outline of solution**

I added a verification on the response to show a "No results found" message if no student or instructor was returned.

### CODE REVIEW

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts [hunk 1]
```diff
@@ -26,6 +26,11 @@ export class AdminSearchPageComponent {
    */
   search(): void {
     this.accountService.searchAccounts(this.searchQuery).subscribe((resp: AdminSearchResult) => {
+      const hasStudents: boolean = !!(resp.students && resp.students.length);
+      const hasInstructors: boolean = !!(resp.instructors && resp.instructors.length);
+      if (!hasStudents && !hasInstructors) {
```
Comment (amrut-prabhu) [inline src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts]: Actually, you can make this an if-else block.
If there are no instructors and no students, then we wouldn't need to hide their links anyways

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts [hunk 2]
```diff
@@ -26,6 +26,11 @@ export class AdminSearchPageComponent {
...
+        this.statusMessageService.showWarningMessage('No results found.');
+      }
       this.instructors = resp.instructors;
       for (const instructor of this.instructors) {
```
Comment (amrut-prabhu) [inline src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts]: While you're at it, can you refactor these to use `hideAllInstructorsLinks()` and `hideAllStudentsLinks()` instead

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts (hunk 1 above)
Comment (Davileal) [reply src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts]: Ok

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts (hunk 2 above)
Comment (Davileal) [reply src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts]: Ok!

**Comment by wkurniawan07** [comment]
> pressing the search button when the search bar is empty shows this (not so user-friendly) error message

Valid issue, although non-blocking. Caused by missed validation for empty search query.

> I assume that this exists in other pages as well.

Not a good assumption...

**Pull request merged**